### PR TITLE
module 6

### PR DESCRIPTION
**Pull request by tilacog**

### CODE REVIEW

**Comment by meaghanlewis** [comment]
Hey @tilacog reminder that I'll review this once I'm back from break the week of Jan 4th.

File: modules-packages-crates/includes/7-knowledge-check.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Knowledge check
+
+Why does the following code won't compile?
```
Comment (meaghanlewis) [inline modules-packages-crates/includes/7-knowledge-check.md]: @tilacog same comment here as in module 5. The question can't be more than a single line. Is there a way to simplify this or have a question without code?

File: modules-packages-crates/includes/1-scoping-terminology.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+A module is a collection of items:
+
+- Constants
+- Type aliases
+- Functions
+- Structs
+- Enums
+- Traits
+- `impl` blocks
+- Other modules, which can nest arbitrarily.
```
Comment (meaghanlewis) [inline modules-packages-crates/includes/1-scoping-terminology.md]: This last point is a bit confusing. I'm not sure I understand about the arbitrary nesting. Can you rephrase?

Maybe it can just say "other modules"

File: modules-packages-crates/includes/1-scoping-terminology.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    │  └── main.rs
+    └── Cargo.toml
+
+A package can have multiple binary crates by placing files in the `src/bin` directory: each file will be a separate binary crate.
+
+If a package contains `src/main.rs` and `src/lib.rs`, it has two crates: a library and a binary, both with the same name as the package.
+
+## Crates
+
+Rust's compilation model centers on artifacts called crates, that can be compiled into a binary or into a library.
```
Comment (meaghanlewis) [inline modules-packages-crates/includes/1-scoping-terminology.md]: Are there any examples of crates you can share? Just thinking of a little more detail to add to this section.

File: modules-packages-crates/includes/1-scoping-terminology.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# Scoping Terminology
```
Comment (meaghanlewis) [inline modules-packages-crates/includes/1-scoping-terminology.md]: Could the title of this unit be changed? Not sure if scoping terminology is descriptive enough. Some suggestions:
- Understand concepts behind code organization
- Understand how Rust programs are organizaed
- Learn more about packags, crates, and modules

**Review by meaghanlewis — COMMENTED** [review]
@tilacog thanks for adding this module! I left a few comments and questions, but otherwise looks good to me.

File: modules-packages-crates/includes/7-knowledge-check.md (hunk 1 above)
Comment (tilacog) [reply modules-packages-crates/includes/7-knowledge-check.md]: I'll keep this discussion open, just for me to keep track of it,  and resolve it the same way as in module 5, then.

**Review by meaghanlewis — APPROVED** [review]
thanks for updating the knowledge check!
